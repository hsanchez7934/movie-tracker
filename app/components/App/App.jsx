import React, { Component } from 'react';
import { getNowPlaying, getMovies } from '../../actions';
import { BrowserRouter, Route, Link } from 'react-router-dom';
import CardCatalogContainer from './../../containers/CardCatalogContainer.js';
import { Switch } from 'react-router';
import Login from './Login/Login';
import Register from './Register/Register';


export default class App extends Component {
  constructor() {
    super();
  }

  render() {
    return (
      <BrowserRouter>
        <div className='app'>
          <Switch>

            <Route exact path='/' render={() => {
              return (
                <div>
                  <Link to={'/login'}>
                    <button>Login</button>
                  </Link>
                  <CardCatalogContainer />
                </div>
              )
            }} />

            <Route exact path='/login' component={Login}/>
            <Route exact path='/register' component={Register} />
          </Switch>

      </div>
      </BrowserRouter>
    )
  }
}
