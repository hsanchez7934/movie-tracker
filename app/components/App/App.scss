.app {
  border: medium dashed green;
}
