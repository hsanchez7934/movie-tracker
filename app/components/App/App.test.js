import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import App from './App';
// import fetchMock from 'fetch-mock';
import renderer from 'react-test-renderer';
import { Route } from 'react-router';
import { shallow, mount, render } from 'enzyme';
import { Router } from 'react-router/MemoryRouter';
import createHistory from 'history/createMemoryHistory';


describe('App', () => {
  it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  });

})
