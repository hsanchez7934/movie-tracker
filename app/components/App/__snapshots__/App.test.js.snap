// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="app"
      >
        <Switch>
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/login"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/register"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/favorites"
            render={[Function]}
          />
        </Switch>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/login"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/register"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/favorites"
            render={[Function]}
          />
        </Switch>,
        "className": "app",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/login"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/register"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/favorites"
              render={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/login",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/register",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/favorites",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="app"
        >
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/login"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/register"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/favorites"
              render={[Function]}
            />
          </Switch>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/login"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/register"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/favorites"
              render={[Function]}
            />
          </Switch>,
          "className": "app",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/login"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/register"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/favorites"
                render={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/login",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/register",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/favorites",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
